transform/transformers/Transformer.py

```diff
@@ -0,0 +1,110 @@
+import logging
+from collections import OrderedDict
+import os.path
+from structlog import wrap_logger
+
+from transform.transformers.CSFormatter import CSFormatter
+from transform.transformers.ImageTransformer import ImageTransformer
+from transform.transformers.survey import Survey
+
+
+class Transformer:
+
+    """A base class for SDX transformers.
+
+    Subclasses should define the contents of the following class variables:
+
+        * :py:const:`defn`
+        * :py:const:`package`
+        * :py:const:`pattern`
+
+    """
+    #: Transformer subclasses declare their transforms in this class variable.
+    #: Each element is a 3-tuple consisting of:
+    #:
+    #: #. An integer or range corresponding to one or more question ids.
+    #: #. A default value for the question(s).
+    #: #. A :ref:`Processing function <processors>`.
+    #:
+    #: Eg, to declare question ids 151, 152, 153 as unsigned integers with a default
+    #: of 0::
+    #:
+    #:  defn = [
+    #:   (range(151, 154, 1), 0, Processor.unsigned_integer),
+    #:
+    #:  ]
+    #:
+    defn = []
+
+    #: Defines a search pattern for survey definitions based on identifiers.
+    #: The path is relative to the location specified by :py:const:`package` above.
+    pattern = "../surveys/{survey_id}.{inst_id}.json"
+
+    receipt_path = os.getenv("SDX_FTP_RECEIPT_PATH", "EDC_QReceipts")
```

Comment:
These are in settings.py, do they need to be done here?

Reply:
changing